File: app/src/main/java/com/instacart/android/challenges/network/NetworkApi.java
package com.instacart.android.challenges.network;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface NetworkApi {

    @GET("orders")
    Call<OrdersResponse> fetchOrders();

    @GET("orders")
    Observable<OrdersResponse> fetchOrdersObservable();

    @GET("order/{order_id}")
    Call<OrderResponse> fetchOrderById(@Path("order_id") long id);

    @GET("order/{order_id}")
    Observable<OrderResponse> fetchOrderByIdObservable(@Path("order_id") long id);

    @GET("discount/{item_id}")
    Call<DiscountResponse> discount(@Path("item_id") long id);

    @GET("discount/{item_id}")
    Observable<DiscountResponse> discountObservable(@Path("item_id") long id);
}

File: app/src/main/java/com/instacart/android/challenges/network/NetworkApi.kt
package com.instacart.android.challenges.network

import io.reactivex.rxjava3.core.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface NetworkApi {

    @GET("order/{order_id}")
    fun fetchOrderItems(@Path("order_id") id: Long): Call<OrderItemsResponse>

    @GET("order/{order_id}")
    fun fetchOrderItemsObservable(@Path("order_id") id: Long): Observable<OrderItemsResponse>

    @GET("order/{order_id}")
    suspend fun fetchOrderItemsCoroutine(@Path("order_id") id: Long): OrderItemsResponse

    @GET("orders")
    fun fetchOrderIds(): Call<OrderIdsResponse>

    @GET("orders")
    fun fetchOrderIdsObservable(): Observable<OrderIdsResponse>

    @GET("orders")
    suspend fun fetchOrderIdsCoroutine(): OrderIdsResponse

    @GET("price/{item_id}")
    fun price(@Path("item_id") id: Long): Call<PriceResponse>

    @GET("price/{item_id}")
    fun priceObservable(@Path("item_id") id: Long): Observable<PriceResponse>

    @GET("price/{item_id}")
    suspend fun priceCoroutine(@Path("item_id") id: Long): PriceResponse
}
